### PR TITLE
Networking stress tests infrastructure improvements.

### PR DESCRIPTION
**Pull request by rzikm**
Key changes:
- Moved common parts of the infra to Common/tests/System/Net/StressTests
- Added support for collecting testhost binaries in case there are crash dumps (enables crash dump investigation)
- Current .NET version is not hardcoded anymore (does not have to be updated every year), is being read from eng/Versions.props

### CODE REVIEW

**Comment by rzikm** [comment]
Stress is running, H/3 stress failure was present before as well.

File: src/libraries/System.Net.Http/tests/StressTests/HttpStress/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ RUN apt-get update -y && \
     apt-get upgrade -y && \
     apt-get install -y cmake clang ruby-dev gem lttng-tools libssl-dev && \
     gem install fpm
-RUN git clone --depth 1 --single-branch --branch v2.3.5 --recursive https://github.com/microsoft/msquic
+RUN git clone --depth 1 --single-branch --branch v2.4.4 --recursive https://github.com/microsoft/msquic
```
Comment (wfurt) [inline src/libraries/System.Net.Http/tests/StressTests/HttpStress/Dockerfile:10]: do we need to build msquic? It feels like we should just consume it from packages.microsoft.com ... and simply get latest e.g. no need to vary about particular varian?

File: eng/docker/libraries-sdk.linux.Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Builds and copies library artifacts into target dotnet sdk image
 ARG BUILD_BASE_IMAGE=mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream9
-ARG SDK_BASE_IMAGE=mcr.microsoft.com/dotnet/nightly/sdk:8.0
+ARG SDK_BASE_IMAGE=mcr.microsoft.com/dotnet/nightly/sdk:latest
```
Comment (wfurt) [inline eng/docker/libraries-sdk.linux.Dockerfile:3]: do we need to worry about cases when "latest" != VERSION

**Review by wfurt — APPROVED** [review]
LGTM. I left few comments.

File: src/libraries/System.Net.Http/tests/StressTests/HttpStress/build-local.sh [hunk 3]
```diff
@@ -1,67 +1,24 @@
...
-    stress_configuration=${stress_configuration^} # Uppercase first character
-fi
+source="${BASH_SOURCE[0]}"
 
-libraries_configuration="Release"
-if [ "$2" != "" ]; then
-    libraries_configuration=${2,,}                      # Lowercase all characters in $1
-    libraries_configuration=${libraries_configuration^} # Uppercase first character
-fi
+# resolve $source until the file is no longer a symlink
```
Comment (antonfirsov) [inline src/libraries/System.Net.Http/tests/StressTests/HttpStress/build-local.sh:12]: Why is this needed?

**Review by antonfirsov — COMMENTED** [review]
SslStress `build-local.ps1` has been left unchanged, we need to update it.

Otherwise looks good, great work!

File: eng/docker/libraries-sdk.linux.Dockerfile (hunk 2 above)
Comment (rzikm) [reply eng/docker/libraries-sdk.linux.Dockerfile:3]: not really, we download the nightly SDK anyway for build purposes (there is no docker image for that). "latest" should guarantee we have all necessary prerequisites (I didn't want to use runtime-deps image because it does not have wget)

File: src/libraries/System.Net.Http/tests/StressTests/HttpStress/build-local.sh (hunk 3 above)
Comment (rzikm) [reply src/libraries/System.Net.Http/tests/StressTests/HttpStress/build-local.sh:12]: It was present in the original scripts, I don't know what the original problem it was solving was.

File: src/libraries/System.Net.Http/tests/StressTests/HttpStress/Dockerfile (hunk 1 above)
Comment (wfurt) [reply src/libraries/System.Net.Http/tests/StressTests/HttpStress/Dockerfile:10]: note may be useful. It was not obvious to me. I'mwondering if `libsmquic-dbg` package would be useful to anybody besides us...

**Comment by rzikm** [comment]
Test failures are unrelated (this PR does not change anything other than http and ssl stress pipelines
